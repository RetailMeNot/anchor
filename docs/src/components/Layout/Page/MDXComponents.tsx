/*
    These are components that will be fed into the MDXProvider.
    https://mdxjs.com/advanced/components#mdxprovider
*/

import * as React from 'react';
import styled from '@xstyled/styled-components';
// ANCHOR & COMPONENTS
import { Typography } from '@retailmenot/anchor';
import { CodePreview } from '../../CodePreview';
import { InlineCodeStyle } from '../../Utils';

const StyledInlineCode = styled('span')`
    ${InlineCodeStyle};
`;

const StyledLi = styled('li')`
    padding-bottom: 0.5rem;
`;

const StyledHr = styled('hr')`
    margin: 2rem 0;
    color: borders.base;
    background-color: borders.base;
    border-style: none;
    height: 1px;
`;

const Components: any = {};

Components.code = (props: any) => <CodePreview {...props} />;
Components.wrapper = (props: any) => <React.Fragment {...props} />;
Components.inlineCode = (props: any) => <StyledInlineCode {...props} />;
Components.pre = (props: any) => <Typography as="pre" {...props} />;
Components.p = (props: any) => <Typography as="p" pb="4" {...props} />;
Components.a = (props: any) => (
    <Typography color="primary.base" as="a" {...props} />
);
Components.blockquote = (props: any) => (
    <Typography as="blockquote" {...props} />
);
Components.strong = (props: any) => (
    <Typography weight={600} as="strong" {...props} />
);
Components.ul = (props: any) => <ul {...props} />;
Components.ol = (props: any) => <ol {...props} />;
Components.li = (props: any) => (
    <StyledLi>
        <Typography {...props} />
    </StyledLi>
);
Components.hr = (props: any) => <StyledHr {...props} />;

Components.h1 = ({ children, ...props }: any) => (
    <Typography
        id={`${children}`.toLowerCase()}
        weight={600}
        as="h1"
        {...props}
        children={children}
        pb="3"
    />
);
Components.h2 = ({ children, ...props }: any) => (
    <>
        <Typography
            as="a"
            name={children
                .split(' ')
                .join('-')
                .toLowerCase()}
        />
        <Typography
            id={`${children.split(' ').join('-')}`.toLowerCase()}
            weight={600}
            as="h2"
            lineHeight={2.3}
            {...props}
            children={children}
            pb="3"
        />
    </>
);
Components.h3 = ({ children, ...props }: any) => (
    <>
        <Typography
            as="a"
            name={children
                .split(' ')
                .join('-')
                .toLowerCase()}
        />
        <Typography
            id={`${children.split(' ').join('-')}`.toLowerCase()}
            weight={600}
            as="h3"
            lineHeight={2.5}
            {...props}
            children={children}
            pb="2"
        />
    </>
);
Components.h4 = ({ children, ...props }: any) => (
    <>
        <Typography
            as="a"
            name={children
                .split(' ')
                .join('-')
                .toLowerCase()}
        />
        <Typography
            id={`${children.split(' ').join('-')}`.toLowerCase()}
            weight={600}
            as="h4"
            {...props}
            children={children}
        />
    </>
);
Components.h5 = ({ children, ...props }: any) => (
    <>
        <Typography
            as="a"
            name={children
                .split(' ')
                .join('-')
                .toLowerCase()}
        />
        <Typography
            id={`${children.split(' ').join('-')}`.toLowerCase()}
            weight={600}
            as="h5"
            {...props}
            children={children}
        />
    </>
);
Components.h6 = ({ children, ...props }: any) => (
    <>
        <Typography
            as="a"
            name={children
                .split(' ')
                .join('-')
                .toLowerCase()}
        />
        <Typography
            id={`${children.split(' ').join('-')}`.toLowerCase()}
            weight={600}
            as="h6"
            mb="1"
            {...props}
            children={children}
        />
    </>
);

export const MDXComponents = Components;
