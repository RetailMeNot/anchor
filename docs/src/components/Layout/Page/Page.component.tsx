// REACT
import * as React from 'react';
// VENDOR
import styled from 'styled-components';
import classNames from 'classnames';
import { MDXProvider } from '@mdx-js/tag';
// COMPONENTS
import { NormalizeCSS, GlobalCSS, sizes } from '../../../../../src/theme';
import { Header, Footer } from '../';
import { SideNavigation } from '../../Navigation';
import { CodePreview } from '../../CodePreview';

export const StyledPageElement = styled.div`
  height: 100vh;
  overflow-y: auto;
  overflow-x: hidden;

  header {
    width: 100%;
    position: sticky;
    top: 0;
  }
`;

interface PageProps {
  className?: string;
  children?: any;
  enableFooter?: boolean;
}

const StyledContentBody = styled.div`
  display: flex;
  height: calc(100vh - 4rem);
  overflow: hidden;


  main {
    overflow-x: hidden;
    overflow-y: auto;
    max-width: 1200px;
    margin: 0 auto;
    padding: ${sizes.padding.lg};
  }
`;

const Components: any = {};

Components.code = (props: any) => <CodePreview {...props} />;
Components.inlineCode = (props: any) => <code {...props} />;
Components.wrapper = (props: any) => <React.Fragment {...props} />;
Components.h1 = (props: any) => <h1 {...props} />;
Components.h2 = (props: any) => <h2 {...props} />;
Components.h3 = (props: any) => <h3 {...props} />;
Components.h4 = (props: any) => <h4 {...props} />;
Components.h5 = (props: any) => <h5 {...props} />;
Components.h6 = (props: any) => <h6 {...props} />;
Components.p = (props: any) => <p {...props} />;
Components.ul = (props: any) => <ul {...props} />;
Components.ol = (props: any) => <ol {...props} />;
Components.li = (props: any) => <li {...props} />;
Components.a = (props: any) => <a {...props} />;
Components.blockquote = (props: any) => <blockquote{...props} />;
Components.strong = (props: any) => <strong {...props} />;

export const Page = ({ children, className, enableFooter }: PageProps): React.ReactElement<any> => (
  <StyledPageElement className={classNames(className)}>
    <NormalizeCSS />
    <GlobalCSS />
    <Header/>
    <StyledContentBody>
      <SideNavigation />
      <main>
        {/* TODO: Add bread-crumbing */}
        <MDXProvider components={Components}>
          {children}
        </MDXProvider>
        {enableFooter && <Footer />}
      </main>
    </StyledContentBody>
  </StyledPageElement>
);

export default Page;
