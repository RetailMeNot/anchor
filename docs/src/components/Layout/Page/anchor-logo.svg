<svg width="150" height="30" xmlns="http://www.w3.org/2000/svg"><defs><linearGradient x1="0%" y1="0%" y2="100%" id="a"><stop stop-color="#DF206C" offset="0%"/><stop stop-color="#DF916C" offset="100%"/></linearGradient></defs><g fill="none" fill-rule="evenodd"><path fill="#FFF" d="M-21-22h1440v3731H-21z"/><path d="M25.242 0h-9.34v6.567c0 1.18.888 2.21 2.066 2.286a2.214 2.214 0 0 0 2.361-2.21V4.43h2.753c.95 0 1.718.768 1.718 1.716v10.989c0 1.674-2.024 2.512-3.206 1.328L8.916 5.785a.794.794 0 0 1 .562-1.356h2.699V0H3.974A3.974 3.974 0 0 0 0 3.975v8.227h4.427V9.479c0-.708.857-1.063 1.358-.562l12.677 12.677c1.183 1.183.345 3.207-1.33 3.207H6.145c-.95 0-1.718-.769-1.718-1.717v-2.752h2.14c1.178 0 2.209-.89 2.286-2.067a2.216 2.216 0 0 0-2.212-2.362H0v9.34a3.985 3.985 0 0 0 3.986 3.987h21.26c2.2 0 3.984-1.784 3.984-3.984V3.986A3.987 3.987 0 0 0 25.242 0" fill="url(#a)"/><path d="M53.563 15.659L51.9 11.517l-1.666 4.142h3.329zm1.071 2.699h-5.472l-1.017 2.515h-3.734L50.123 7.93h3.607l5.73 12.942h-3.808l-1.018-2.515zM76.437 7.93v12.943h-3.014l-5.712-6.897v6.897h-3.587V7.93h3.013l5.713 6.896V7.93h3.587M85.49 20.304a6.428 6.428 0 0 1-2.541-2.394c-.617-1.023-.925-2.182-.925-3.476s.308-2.453.925-3.476a6.432 6.432 0 0 1 2.542-2.394c1.078-.574 2.295-.86 3.65-.86 1.185 0 2.25.21 3.2.628a6.144 6.144 0 0 1 2.365 1.812l-2.33 2.108c-.837-1.01-1.855-1.516-3.049-1.516-.702 0-1.328.154-1.876.462a3.206 3.206 0 0 0-1.276 1.304c-.303.56-.453 1.205-.453 1.932 0 .727.15 1.371.453 1.932.302.56.728.995 1.276 1.303.548.309 1.174.462 1.876.462 1.194 0 2.212-.505 3.05-1.516l2.33 2.108a6.14 6.14 0 0 1-2.366 1.812c-.95.419-2.015.629-3.2.629-1.355 0-2.572-.287-3.65-.86M112.466 7.93v12.943h-3.662v-5.048h-4.991v5.048h-3.662V7.93h3.662v4.862h4.991V7.931h3.662M127.02 17.67a3.333 3.333 0 0 0 1.255-1.304c.308-.56.462-1.205.462-1.932 0-.727-.154-1.371-.462-1.932a3.336 3.336 0 0 0-1.255-1.304 3.468 3.468 0 0 0-1.777-.462c-.652 0-1.244.154-1.774.462-.53.308-.95.743-1.258 1.304-.308.56-.462 1.205-.462 1.932 0 .727.154 1.371.462 1.932.309.56.728.995 1.258 1.303.53.309 1.122.462 1.774.462.655 0 1.246-.153 1.777-.462zm-5.466 2.625a6.536 6.536 0 0 1-2.57-2.404c-.622-1.023-.933-2.175-.933-3.457s.31-2.434.934-3.457a6.54 6.54 0 0 1 2.57-2.404c1.092-.58 2.32-.87 3.688-.87 1.369 0 2.6.29 3.689.87a6.524 6.524 0 0 1 2.57 2.404c.623 1.023.933 2.175.933 3.457s-.31 2.434-.933 3.457a6.52 6.52 0 0 1-2.57 2.404c-1.09.58-2.32.869-3.689.869-1.368 0-2.596-.29-3.689-.87zM145.387 11.314c-.383-.333-.95-.5-1.702-.5h-2.035v3.79h2.035c.752 0 1.319-.162 1.702-.489.38-.327.573-.792.573-1.396 0-.604-.193-1.072-.573-1.405zm-1.738 6.12h-1.999v3.439h-3.66V7.93h5.917c1.172 0 2.187.194 3.05.582.863.388 1.53.943 1.998 1.664.467.721.703 1.569.703 2.542 0 .937-.22 1.754-.657 2.45-.438.696-1.063 1.242-1.876 1.636l2.791 4.068h-3.92l-2.347-3.44z" fill="#323232"/></g></svg>
