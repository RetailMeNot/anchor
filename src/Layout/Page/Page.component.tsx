// REACT
import * as React from 'react';
// VENDOR
import styled from 'styled-components';
// import { Box } from '@rebass/grid';
// STYLES
import { colors, sizes } from '../../theme';
// COMPONENTS
import { Footer, Header } from '../';
import { NormalizeCSS, GlobalCSS } from '../../theme/GlobalStyles';

export const StyledPageElement = styled.section`
  display: flex;
  flex-direction: column;
  height: 100vh;
  max-width: 100vw;

  header {
    flex: 0 0 auto;
    max-width: 100vw;
  }

  footer {
    flex: 0 0 auto;
    max-width: 100vw;
  }
`;

export const StyledBodyElement = styled.section`
  flex: 1 0 auto;
  background-color: ${colors.neutrals.silver.base};
  max-width: 100vw;
  align-self: stretch;
  padding: ${sizes.padding.lg} ${sizes.padding.xl};
`;

interface PageProps {
  className?: string;
  children?: any;
}

export const Page = ({ children }: PageProps): React.ReactElement<any> => (
  <StyledPageElement>
    <NormalizeCSS />
    <GlobalCSS />
    <Header />
    <StyledBodyElement>
      {children}
    </StyledBodyElement>
    <Footer />
  </StyledPageElement>
);
