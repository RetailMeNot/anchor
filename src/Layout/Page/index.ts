export { Page } from './Page.component';
