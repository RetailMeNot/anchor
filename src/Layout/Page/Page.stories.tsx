// REACT
import * as React from 'react';
// STORYBOOK
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
// COMPONENT
import { Page } from './Page.component';

const StyledStory = styled.div`
  padding: 10rem;
`;

storiesOf('Components/Page', module).add('default', () => (
    <StyledStory>
        <Page>Story</Page>
    </StyledStory>
));
