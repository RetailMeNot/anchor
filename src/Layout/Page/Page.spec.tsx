// REACT
import * as React from 'react';
import * as renderer from 'react-test-renderer';
// VENDOR
import { shallow, mount } from 'enzyme';
// COMPONENT
import Page, {
  Page as PageComponent,
  DEFAULT_HEADER_HEIGHT,
  DEFAULT_FOOTER_HEIGHT,
  DEFAULT_LAYOUT_WIDTH,
  TRANSPARENT,
  DEFAULT_CONTENT_WIDTH,
  DEFAULT_SIDEBAR_WIDTH,
  LEFT,
  RIGHT
} from './Page.component';

describe('Component: Page', () => {
  const subject = <PageComponent />;
  const wrapper = mount(subject);
  const component = shallow(subject);

  it('should be defined.', () => {
    expect(subject).toBeDefined();
    expect(component).toBeDefined();

    const tree = renderer.create(subject).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render with default settings', () => {
    expect(component.find('.anchor-page').exists()).toBeTruthy();
    expect(wrapper.find('Grid__Grid').prop('rows'))
      .toEqual(`minmax(${DEFAULT_HEADER_HEIGHT},auto) 1fr minmax(${DEFAULT_FOOTER_HEIGHT},auto)`);
  });

  it('should render with passed props.', () => {
    const header = <header>Test Header</header>;
    const footer = <footer>Test Footer</footer>;
    const footerHeight = '2rem';
    const headerHeight = '8rem';
    const testSubject = shallow(
      <PageComponent headerHeight={headerHeight} footerHeight={footerHeight} header={header} footer={footer} />
    );

    expect(testSubject.find('header').exists()).toBeTruthy();
    expect(testSubject.find('footer').exists()).toBeTruthy();
    expect(testSubject.find('Grid__Grid').prop('rows'))
      .toEqual(`minmax(${headerHeight},auto) 1fr minmax(${footerHeight},auto)`);
  });
});

describe('Component: DefaultLayout', () => {
  const subject = <Page.DefaultLayout />;
  const wrapper = mount(subject);
  const component = shallow(subject);

  it('should be defined.', () => {
    expect(subject).toBeDefined();
    expect(component).toBeDefined();

    const tree = renderer.create(subject).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render with default settings.', () => {
    expect(component.find('.anchor-default-layout').exists()).toBeTruthy();
    expect(wrapper.find('DefaultLayout').prop('layoutWidth'))
      .toEqual(DEFAULT_LAYOUT_WIDTH);
    expect(wrapper.find('DefaultLayout').prop('layoutBackgroundColor'))
      .toEqual(TRANSPARENT);
    expect(wrapper.find('DefaultLayout').prop('contentWidth'))
      .toEqual(DEFAULT_CONTENT_WIDTH);
    expect(wrapper.find('DefaultLayout').prop('contentBackgroundColor'))
      .toEqual(TRANSPARENT);
  });

  it('should render with passed props.', () => {
    const layoutWidth = '80px';
    const layoutBackgroundColor = '#ffffff';
    const contentWidth = '2rem';
    const contentBackgroundColor = 'orange';
    const testSubject = mount(
      <Page.DefaultLayout
        layoutWidth={layoutWidth}
        layoutBackgroundColor={layoutBackgroundColor}
        contentWidth={contentWidth}
        contentBackgroundColor={contentBackgroundColor} />
    );

    expect(testSubject.find('DefaultLayout').prop('layoutWidth'))
      .toEqual(layoutWidth);
    expect(testSubject.find('DefaultLayout').prop('layoutBackgroundColor'))
      .toEqual(layoutBackgroundColor);
    expect(testSubject.find('DefaultLayout').prop('contentWidth'))
      .toEqual(contentWidth);
    expect(testSubject.find('DefaultLayout').prop('contentBackgroundColor'))
      .toEqual(contentBackgroundColor);
  });
});

describe('Component: SidebarLayout', () => {
  const subject = <Page.SidebarLayout />;
  const wrapper = mount(subject);
  const component = shallow(subject);

  it('should be defined.', () => {
    expect(subject).toBeDefined();
    expect(component).toBeDefined();

    const tree = renderer.create(subject).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render with default settings.', () => {
    expect(component.find('.anchor-sidebar-layout').exists()).toBeTruthy();

    // tslint:disable-next-line: no-console
    console.log(wrapper.debug());

    expect(wrapper.find('SidebarLayout').prop('layoutWidth'))
      .toEqual(DEFAULT_CONTENT_WIDTH);
    expect(wrapper.find('SidebarLayout').prop('layoutBackgroundColor'))
      .toEqual(TRANSPARENT);
    expect(wrapper.find('SidebarLayout').prop('contentWidth'))
      .toEqual(DEFAULT_CONTENT_WIDTH);
    expect(wrapper.find('SidebarLayout').prop('contentBackgroundColor'))
      .toEqual(TRANSPARENT);
    expect(wrapper.find('SidebarLayout').prop('sidebarWidth'))
      .toEqual(DEFAULT_SIDEBAR_WIDTH);
    expect(wrapper.find('SidebarLayout').prop('sidebarAlign'))
      .toEqual(LEFT);
  });

  it('should render with passed props.', () => {
    const layoutWidth = '80px';
    const layoutBackgroundColor = '#ffffff';
    const contentWidth = '2rem';
    const contentBackgroundColor = 'orange';
    const sidebarWidth = '5.5rem';
    const testSubject = mount(
      <Page.SidebarLayout
        layoutWidth={layoutWidth}
        layoutBackgroundColor={layoutBackgroundColor}
        contentWidth={contentWidth}
        contentBackgroundColor={contentBackgroundColor}
        sidebarWidth={sidebarWidth}
        sidebarAlign={RIGHT}
      />
    );

    expect(testSubject.find('SidebarLayout').prop('layoutWidth'))
      .toEqual(layoutWidth);
    expect(testSubject.find('SidebarLayout').prop('layoutBackgroundColor'))
      .toEqual(layoutBackgroundColor);
    expect(testSubject.find('SidebarLayout').prop('contentWidth'))
      .toEqual(contentWidth);
    expect(testSubject.find('SidebarLayout').prop('contentBackgroundColor'))
      .toEqual(contentBackgroundColor);
    expect(testSubject.find('SidebarLayout').prop('sidebarWidth'))
      .toEqual(sidebarWidth);
    expect(testSubject.find('SidebarLayout').prop('sidebarAlign'))
      .toEqual(RIGHT);
  });
});

    // tslint:disable-next-line: no-console
    // console.log(wrapper.debug());