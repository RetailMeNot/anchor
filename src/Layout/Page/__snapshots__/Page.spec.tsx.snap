// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Page should be defined. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  max-width: 100%;
  width: 100vw;
}

<main
  className="c0 anchor-page"
>
  <div>
    <div />
  </div>
</main>
`;
